airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -136,34 +147,92 @@ def _create_prepared_request(
+
+        This method uses the retry logic to send the request. It determines the maximum number of retries
+        by checking if the error handler or any backoff strategy has a defined max_retries property.
+        If not, it falls back to the default value.
+
+        The `backoff` package uses the `max_tries` parameter, which indicates the total number of
+        attempts before giving up. If this number is 0, no calls are expected to be made. However,
+        this method uses `max_retries` assuming there would be at least one initial attempt and some
+        retry attempts, so it adds 1 to the expected retry attempts.
+
+        The method also determines the maximum time for retries and the factor for exponential backoff
+        based on the error handler and backoff strategies.
+
+        Args:
+            request (requests.PreparedRequest): The prepared HTTP request to send.
+            request_kwargs (Mapping[str, Any]): Additional keyword arguments for the request.
+
+        Returns:
+            requests.Response: The HTTP response received from the server after retries.
         """
-        max_retries = self._backoff_strategy.max_retries or self._DEFAULT_MAX_RETRY  # type: ignore # max_retries is included in default implemention but optional
-        max_tries = max(0, max_retries) + 1
 
-        max_time = self._backoff_strategy.max_time or self._DEFAULT_MAX_TIME  # type: ignore # max_time is included in default implemention but optional
+        max_retries = self._max_retries()
+        max_tries = max(0, max_retries) + 1
+        max_time = self._max_time()
+        factor = self._factor()
 
         user_backoff_handler = user_defined_backoff_handler(max_tries=max_tries, max_time=max_time)(self._send)
-        backoff_handler = http_client_default_backoff_handler(max_tries=max_tries, max_time=max_time, factor=self._DEFAULT_RETRY_FACTOR)
+        backoff_handler = http_client_default_backoff_handler(max_tries=max_tries, max_time=max_time, factor=factor)
         # backoff handlers wrap _send, so it will always return a response
-        response = backoff_handler(user_backoff_handler)(request, request_kwargs)
+        response = backoff_handler(user_backoff_handler)(request, request_kwargs, log_formatter=log_formatter)  # type: ignore # mypy can't infer that backoff_handler wraps _send
 
         return response
 
-    def _send(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+    def _send(
+        self,
+        request: requests.PreparedRequest,
+        request_kwargs: Mapping[str, Any],
+        log_formatter: Optional[Callable[[requests.Response], Any]] = None,
+    ) -> requests.Response:
 
         if request not in self._request_attempt_count:
             self._request_attempt_count[request] = 1
         else:
             self._request_attempt_count[request] += 1
 
-        self._logger.debug(
+        self._logger.info(
```

Comment:
Why are you changing the log level? Logging all requests may be too noisy

Reply:
Nice catch. This was a mistake.